packages/design/src/FormManager/FormList/CreateNew/index.tsx

```diff
@@ -13,70 +14,162 @@ export default function CreateNew() {
     createNewForm: state.createNewForm,
     createNewFormByPDFUpload: state.createNewFormByPDFUpload,
     createNewFormByPDFUrl: state.createNewFormByPDFUrl,
+    addNotification: state.addNotification,
   }));
+
+  const handleSampleDocumentImport = async (url: string) => {
+    try {
+      const result = await actions.createNewFormByPDFUrl(url);
+      if (result.success) {
+        navigate(`/${result.data}/create`, {
+          state: {
+            result,
+          },
+        });
+      } else {
+        // show error toast
+        actions.addNotification(
+          'error',
+          'Sorry, but there was an error importing the form.'
+        );
+      }
+    } catch (e: unknown) {
+      // show error toast
+      if ((e as Error).message) {
+        actions.addNotification('error', (e as Error).message);
```

Comment:
To keep the view free of logic, could the error handler with notifications be added to `createNewFormByPDFUrl`, and just leave the `navigate` call?